source/_patterns/01-atoms/icons/icon.twig

```diff
@@ -1,9 +1,17 @@
+{% if icon.width and icon.height %}
+  {% set height_width = 'width-' ~ icon.width ~ ' height-' ~ icon.height %}
+{% elseif icon.width %}
+  {% set height_width = 'width-' ~ icon.width ~ ' height-' ~ icon.width %}
+{% elseif icon.height %}
+  {% set height_width = 'width-' ~ icon.height ~ ' height-' ~ icon.height %}
+{% endif %}
+
 {% if icon.name %}
   {% spaceless %}
-    <svg class="icon icon-{{ icon.name }}" role="img"
+    <svg class="icon icon-{{ icon.name }} {%- if height_width %} {{ height_width }}{% endif -%}" role="img"
       {%- if icon.title %} title="{{ icon.title }}"{% endif -%}
-      {%- if decorative %} aria-hidden="true"{% endif -%}>
-      <use xlink:href="{{ icon.file }}#i-{{ icon.name }}"></use>
+      {%- if decorative %} aria-hidden="true"{% endif -%}">
+      <use {% if icon.color %} class="text-{{ icon.color }}"{% endif %} xlink:href="{{ icon.file }}#i-{{ icon.name }}"></use>
```

Comment:
try this:
class="{{ icon.color|default('') }}"
instead of:
{% if icon.color %} class="text-{{ icon.color }}"{% endif %}